### PR TITLE
Find assets - filter by createdBy

### PR DESCRIPTION
**Pull request by vanruch**

### CODE REVIEW

File: src/services/entity_builder.js [hunk 1]
```diff
@@ -149,6 +149,23 @@ export default class EntityBuilder {
     return {...params};
   }
 
+  validateAndCastFindAssetsParams(params) {
+    const allowedParametersList = ['page', 'perPage', 'createdBy'];
+
+    validateFieldsConstrainedToSet(params, allowedParametersList);
+
+    this.ensureGeoLocationParamsCorrectlyPlaced(params);
```
Comment (Anze1m) [inline src/services/entity_builder.js]: No data field in asset = no geo data field in asset.

File: test/services/entity_builder.js [hunk 2]
```diff
@@ -376,10 +376,36 @@ describe('Entity Builder', () => {
     });
   });
 
-  describe('Validating query parameters', () => {
+  describe('Validating query parameters - Event', () => {
```
Comment (Anze1m) [inline test/services/entity_builder.js]: Asset not event

File: test/services/entity_builder.js [hunk 3]
```diff
@@ -376,10 +376,36 @@ describe('Entity Builder', () => {
...
+      const params = put(validParamsAsStrings, 'createdBy', '0x12312312');
+      expect(() => entityBuilder.validateAndCastFindEventsParams(params)).to.throw(ValidationError);
+    });
+  });
+
+  describe('Validating query parameters - Event', () => {
+    let entityBuilder;
+    const anAddress = '0xEaE0D78450DaB377376206f419E9bCA0D28829F9';
+    const validParamsAsStrings = {assetId: '0x1234', fromTimestamp: '10', toTimestamp: '20', page: '2', perPage: '4',
+      createdBy: anAddress};
```
Comment (Anze1m) [inline test/services/entity_builder.js]: Why not?
```
const validParamsAsStrings = {
    assetId: '0x1234',
    fromTimestamp: '10',
    toTimestamp: '20',
    page: '2',
    perPage: '4',
    createdBy: anAddress
};
```

File: src/services/entity_builder.js (hunk 1 above)
Comment (vanruch) [reply src/services/entity_builder.js]: 👍

File: test/services/entity_builder.js (hunk 2 above)
Comment (vanruch) [reply test/services/entity_builder.js]: 👍

File: src/services/entity_builder.js [hunk 4]
```diff
@@ -149,6 +149,21 @@ export default class EntityBuilder {
...
+
+    params.page = validateIntegerParameterAndCast(params.page, 'page');
+    params.perPage = validateIntegerParameterAndCast(params.perPage, 'perPage');
+    if (params.createdBy) {
+      validateIsAddress(params.createdBy);
+    }
+
+    return {...params};
+  }
+
```
Comment (Anze1m) [inline src/services/entity_builder.js]: We already got method 'validateAndCastFindEventsParams' that is really, really similar.
```
  validateAndCastFindEventsParams(params) {
    const allowedParametersList = ['assetId', 'fromTimestamp', 'toTimestamp', 'page', 'perPage', 'createdBy', 'data'];

    validateFieldsConstrainedToSet(params, allowedParametersList);

    this.ensureGeoLocationParamsCorrectlyPlaced(params);

    params.fromTimestamp = validateIntegerParameterAndCast(params.fromTimestamp, 'fromTimestamp');
    params.toTimestamp = validateIntegerParameterAndCast(params.toTimestamp, 'toTimestamp');
    params.page = validateIntegerParameterAndCast(params.page, 'page');
    params.perPage = validateIntegerParameterAndCast(params.perPage, 'perPage');
    if (params.createdBy) {
      validateIsAddress(params.createdBy);
    }

    return {...params};
  }
```
We also have something similar for account querying.
```
  validateAndCastFindAccountParams(params) {
    const allowedParametersList = ['accessLevel', 'page', 'perPage', 'registeredBy'];

    validateFieldsConstrainedToSet(params, allowedParametersList);

    params.accessLevel = validateIntegerParameterAndCast(params.accessLevel, 'accessLevel');
    params.page = validateIntegerParameterAndCast(params.page, 'page');
    params.perPage = validateIntegerParameterAndCast(params.perPage, 'perPage');
    if (params.registeredBy) {
      validateIsAddress(params.registeredBy);
    }

    return {...params};
  }
```
Maybe we should think about moving some stuff from 'EntityBuilder' and 'AccountAccessDefinitions' to some kind of query utils, or query validators?

Comment (marekkirejczyk) [reply src/services/entity_builder.js]: Shall we add a sperate ticket for that?
Sth like:
```js
validateAndCastFindEventsParams(params) {
  validate(params, {
    page: integer, 
    perPage: integer, 
    createdBy: {type: address, required: false}
  });
}
```

**Pull request merged**